azure-keyvault/src/main/java/com/microsoft/azure/keyvault/authentication/KeyVaultCredentials.java

```diff
@@ -96,6 +103,23 @@ public Response intercept(Chain chain) throws IOException {
             Map<String, String> challengeMap) throws IOException {
 
         Boolean supportsPop = supportsMessageProtection(originalRequest.url().toString(), challengeMap);
+
+        // if the service supports pop and a clientEncryptionKey has not been generated yet, generate
+        // the key that will be used for encryption on this an all subsequent protected requests

```

Comment:
Nit: typo; should be and
```suggestion
        // the key that will be used for encryption on this and all subsequent protected requests
```